tests/extension/khr_queue_empty_query/queue_empty.cpp

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2025 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include <atomic>
+#include <iostream>
+
+namespace queue_empty_query::tests {
+
+TEST_CASE(
+    "the queue empty query extension defines the "
+    "SYCL_KHR_QUEUE_EMPTY_QUERY macro",
+    "[khr_queue_empty_query]") {
+#ifndef SYCL_KHR_QUEUE_EMPTY_QUERY
+  static_assert(false, "SYCL_KHR_QUEUE_EMPTY_QUERY is not defined");
+#endif
+}
+
+TEST_CASE("queue are empty by default", "[khr_queue_empty_query]") {
+  sycl::queue q{};
+  CHECK(q.khr_empty());
+}
+
+TEST_CASE("queue are not empty when a command have been enqueed",
+          "[khr_queue_empty_query]") {
+  sycl::queue q{};
+  std::atomic_bool start = false;
+
+  auto e1 = q.submit([&](sycl::handler& cgh) {
+    cgh.host_task([&]() {
+      // To avoid the loop being optimized away
+      int iter = 0;
```

Comment:
```suggestion
      unsigned int iter = 0;
```
To avoid UB caused by (theoretical) signed integer overflow.

Reply:
Good catch! On the same idea, are you ok with `printf` in the CTS, or should I try to do something else?

Reply:
You are right. Printing some random value w/o any context doesn't seem nice. Consider using something like https://github.com/KhronosGroup/SYCL-CTS/blob/f7a26596b2085d40982ae7f94cee729ec5a73d86/tests/event/event.h#L29. 
Alternatively, I suppose just writing `iter` into some memory created outside of the host_task scope should be enough to prevent compiler optimizations.

Reply:
Oh amazing. I never used `std::promise`; far better, not more ugly atomic and ugly spin-lock. Thanks!